arroyo-connectors/src/filesystem.rs

```diff
@@ -0,0 +1,191 @@
+            Some(FormatSettings::Json {  }) => ("FileSystem<JSON>".to_string(), "connectors::filesystem::JsonFileSystemSink::<#in_k, #in_t>"),
+            None => bail!("have to have some format settings"),
+        };
+
+        let config = OperatorConfig {
+            connection: serde_json::to_value(config).unwrap(),
+            table: serde_json::to_value(table).unwrap(),
+            rate_limit: None,
+            serialization_mode: Some(serialization_mode(schema.as_ref().unwrap())),
+        };
+
+        Ok(Connection {
+            id,
+            name: name.to_string(),
+            connection_type: ConnectionType::Sink,
+            schema: schema
+                .map(|s| s.to_owned())
+                .ok_or_else(|| anyhow!("No schema defined for FileSystemSink"))?,
+            operator: operator.to_string(),
+            config: serde_json::to_string(&config).unwrap(),
+            description,
+        })
+    }
+
+    fn from_options(
+        &self,
+        name: &str,
+        opts: &mut std::collections::HashMap<String, String>,
+        schema: Option<&ConnectionSchema>,
+    ) -> anyhow::Result<crate::Connection> {
+        let write_target = if let (Some(s3_bucket), Some(s3_directory), Some(aws_region)) = (
+            opts.remove("s3_bucket"),
+            opts.remove("s3_directory"),
+            opts.remove("aws_region"),
+        ) {
+            Destination::S3Bucket {
+                s3_bucket,
+                s3_directory,
+                aws_region,
+            }
+        } else {
+            bail!("Only s3 targets are supported currently")
+        };
+
+        let inactivity_rollover_seconds = opts
+            .remove("inactivity_rollover_seconds")
+            .map(|value| value.parse::<i64>().map_err(|e| anyhow::anyhow!(e)))
+            .transpose()?;
+        let max_parts = opts
+            .remove("max_parts")
+            .map(|value| value.parse::<i64>().map_err(|e| anyhow::anyhow!(e)))
+            .transpose()?;
+        let rollover_seconds = opts
+            .remove("rollover_seconds")
+            .map(|value| value.parse::<i64>().map_err(|e| anyhow::anyhow!(e)))
+            .transpose()?;
+        let target_file_size = opts
+            .remove("target_file_size")
+            .map(|value| value.parse::<i64>().map_err(|e| anyhow::anyhow!(e)))
+            .transpose()?;
+        let target_part_size = opts
+            .remove("target_part_size")
+            .map(|value| value.parse::<i64>().map_err(|e| anyhow::anyhow!(e)))
+            .transpose()?;
+
+        let file_settings = Some(FileSettings {
+            inactivity_rollover_seconds,
+            max_parts,
+            rollover_seconds,
+            target_file_size,
+            target_part_size,
+        });
+        let format_settings = match pull_opt("fileformat", opts)?.as_str() {
```

Comment:
I think it's confusing that all other connectors are configured using `format`, but here it's `fileformat`